### DISCUSSION TITLE
How to change SSL CTX Cipher and MAC implementations to different provider after it created.

### DISCUSSION THREAD

**Discussion by jsrikanth1608** [openssl/openssl#25286] [Q&A] [answered]
The function "ssl_load_ciphers" is resolving the Cipher and MAC implementations from underlying available providers (ex: default) while creating the SSL CTX. In our case the HW providers are loaded after some of the initial SSL CTX creations. Please let me know how to change the Cipher and MAC implementations of these SSL CTX to use newly loaded HW provider instead of default.

Thank you.

**Comment by mattcaswell** [comment, marked as answer]
This is not currently possible. The recommended approach is to recreate the SSL_CTX in this case.

**Reply by jsrikanth1608** [reply]
Thank you.

**Reply by jsrikanth1608** [reply]
Follow-up on this is, does new SSL connections will inherit the cipher and mac methods from SSL_CTX or try to resolve from the existing providers at that time.?
And I see only the cipher and mac references are storing in SSL_CTX not other methods ex: ECDH, SIGNATURE etc in SSL_CTX_new() function call. Any specific reasons for this. please let me know.
